src/bpftrace.cpp

```diff
@@ -1757,31 +1759,33 @@ std::optional<std::vector<uint8_t>> BPFtrace::find_empty_key(
 }
 
 std::string BPFtrace::get_stack(int64_t stackid,
+                                uint32_t nr_stack_frames,
                                 int pid,
                                 int probe_id,
                                 bool ustack,
                                 StackType stack_type,
                                 int indent)
 {
+  struct stack_key stack_key = { stackid, nr_stack_frames };
   auto stack_trace = std::vector<uint64_t>(stack_type.limit);
   int err = bpf_lookup_elem(bytecode_.getMap(stack_type.name()).fd,
-                            &stackid,
+                            &stack_key,
                             stack_trace.data());
   if (err) {
     // ignore EFAULT errors: eg, kstack used but no kernel stack
-    if (stackid != -EFAULT)
-      LOG(ERROR) << "failed to look up stack id " << stackid << " (pid " << pid
-                 << "): " << err;
+    LOG(ERROR) << "failed to look up stack id: " << stackid
+               << " stack length: " << nr_stack_frames << " (pid " << pid
+               << "): " << err;
     return "";
   }
 
   std::ostringstream stack;
   std::string padding(indent, ' ');
 
   stack << "\n";
-  for (auto &addr : stack_trace) {
-    if (addr == 0)
-      break;
+  uint32_t i;
+  for (i = 0; i < nr_stack_frames; ++i) {
```

Comment:
I don't think that we need to follow kernel's "forward variable declaration" policy:
```suggestion
  for (uint32_t i = 0; i < nr_stack_frames; ++i) {
```